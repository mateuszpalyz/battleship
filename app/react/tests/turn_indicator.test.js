import React from 'react';
import TurnIndicator from '../components/turn_indicator';
import renderer from 'react-test-renderer';

test('Component displays player when it is players turn', () => {
  const component = renderer.create(
    <TurnIndicator turn='player1' player='player1'/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Component displays opponent when it is opponents turn', () => {
  const component = renderer.create(
    <TurnIndicator turn='player2' player='player1'/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
