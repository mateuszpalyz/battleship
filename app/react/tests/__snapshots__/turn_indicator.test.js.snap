// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component displays opponent when it is opponents turn 1`] = `
<div
  className="game-status"
>
  <p>
    Now shooting: 
    Opponent
  </p>
</div>
`;

exports[`Component displays player when it is players turn 1`] = `
<div
  className="game-status"
>
  <p>
    Now shooting: 
    You
  </p>
</div>
`;
